cannot build in venv Debian Trixie

Debian Trixie was recently released as the new stable version of Debian. So I needed to rebuild virtual environments.

```
  Building wheel for rrdtool-bindings (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for rrdtool-bindings (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [167 lines of output]
      running bdist_wheel
      running build
      running build_py
      running egg_info
      writing src/rrdtool_bindings.egg-info/PKG-INFO
      writing dependency_links to src/rrdtool_bindings.egg-info/dependency_links.txt
      writing top-level names to src/rrdtool_bindings.egg-info/top_level.txt
      reading manifest file 'src/rrdtool_bindings.egg-info/SOURCES.txt'
      writing manifest file 'src/rrdtool_bindings.egg-info/SOURCES.txt'
      creating build/lib.linux-x86_64-cpython-313/rrdtool
      copying src/rrdtool/rrdtoolmodule.c -> build/lib.linux-x86_64-cpython-313/rrdtool
      copying src/rrdtool/rrdtoolmodule.h -> build/lib.linux-x86_64-cpython-313/rrdtool
      running build_ext
      building 'rrdtool' extension
      creating build/temp.linux-x86_64-cpython-313/src/rrdtool
      gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/srv/modoboa/env/include -I/usr/include/python3.13 -c src/rrdtool/rrdtoolmodule.c -o build/temp.linux-x86_64-cpython-313/src/rrdtool/rrdtoolmodule.o
      src/rrdtool/rrdtoolmodule.c: In function ‘_rrdtool_create’:
      src/rrdtool/rrdtoolmodule.c:278:39: error: passing argument 2 of ‘rrd_create’ from incompatible pointer type [-Wincompatible-pointer-types]
        278 |     status = rrd_create(rrdtool_argc, rrdtool_argv);
            |                                       ^~~~~~~~~~~~
            |                                       |
            |                                       const char **
      In file included from src/rrdtool/rrdtoolmodule.h:4,
                       from src/rrdtool/rrdtoolmodule.c:27:
      /usr/include/rrd.h:157:5: note: expected ‘char **’ but argument is of type ‘const char **’
        157 |     char **);
            |     ^~~~~~~
      src/rrdtool/rrdtoolmodule.c: In function ‘_rrdtool_dump’:
      src/rrdtool/rrdtoolmodule.c:317:37: error: passing argument 2 of ‘rrd_dump’ from incompatible pointer type [-Wincompatible-pointer-types]
        317 |     status = rrd_dump(rrdtool_argc, rrdtool_argv);
            |                                     ^~~~~~~~~~~~
            |                                     |
            |                                     const char **
      /usr/include/rrd.h:205:5: note: expected ‘char **’ but argument is of type ‘const char **’
        205 |     char **);
            |     ^~~~~~~
      src/rrdtool/rrdtoolmodule.c: In function ‘_rrdtool_update’:
      src/rrdtool/rrdtoolmodule.c:356:39: error: passing argument 2 of ‘rrd_update’ from incompatible pointer type [-Wincompatible-pointer-types]
        356 |     status = rrd_update(rrdtool_argc, rrdtool_argv);
            |                                       ^~~~~~~~~~~~
            |                                       |
            |                                       const char **
      /usr/include/rrd.h:174:5: note: expected ‘char **’ but argument is of type ‘const char **’
        174 |     char **);
            |     ^~~~~~~
      src/rrdtool/rrdtoolmodule.c: In function ‘_rrdtool_updatev’:
      src/rrdtool/rrdtoolmodule.c:389:39: error: passing argument 2 of ‘rrd_update_v’ from incompatible pointer type [-Wincompatible-pointer-types]
        389 |     data = rrd_update_v(rrdtool_argc, rrdtool_argv);
            |                                       ^~~~~~~~~~~~
            |                                       |
            |                                  g
```

--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,9 @@
 """Instructions for setuptools to build the rrdtool bindings."""
 
 import os
+import platform
 import subprocess
+import sys
 
 from setuptools import Extension, setup
 
@@ -9,12 +11,27 @@ from setuptools import Extension, setup
 if subprocess.run(["command -v gcc-11"], shell=True, check=False).returncode > 0:
     os.environ["CC"] = "gcc"
 
+extra_compile_args = []
+
+# Debian Trixie: disable incompatible-pointer-types
+if sys.version_info >= (3, 10):
+    distro_info = platform.freedesktop_os_release()
+    if distro_info["ID"] == "debian" and distro_info["VERSION_ID"] == "13":
+        extra_compile_args.append("-Wno-incompatible-pointer-types")
+elif not subprocess.run(
+    ["/usr/bin/grep", "13", "/etc/debian_version"], check=False
+).returncode:
+    extra_compile_args.append("-Wno-incompatible-pointer-types")
+
+print("Extra compile args:", extra_compile_args)
+
 setup(
     ext_modules=[
         Extension(
             name="rrdtool",
             sources=["src/rrdtool/rrdtoolmodule.c"],
             libraries=["rrd"],
+            extra_compile_args=extra_compile_args,
         )
     ]
 )

